db/db_impl/db_impl.cc

```diff
@@ -4789,6 +4826,44 @@ Status DBImpl::VerifyChecksum(const ReadOptions& read_options) {
   return s;
 }
 
+Status DBImpl::VerifySstFileChecksum(const FileMetaData& fmeta,
+                                     const std::string& fname,
+                                     const std::string& file_checksum_func_name,
+                                     const ReadOptions& read_options) {
+  Status s;
+  if (fmeta.file_checksum == kUnknownFileChecksum) {
+    return s;
+  }
+  if (fmeta.file_checksum_func_name != file_checksum_func_name) {
```

Comment:
This is the wrong approach. A checksum generator factory can support different functions, though it does have a default function. To support migration, it should be OK for the SST files to have a mix of checksum functions, as long as they're all supported by the configured factory. You have to request the function used in the SST file from the generator factory, then verify that the function returned matches what you requested. If that doesn't match, then you have an error.

Reply:
Updated.